build.gradle

```diff
@@ -39,9 +39,34 @@ String getConclaveRevision() {
     return conclave_revision
 }
 
+enum ArtifactType {
+    GA_RELEASE,
+    RELEASE_CANDIDATE,
+    SNAPSHOT
+}
+
+def gitBranch = getGitBranch()
+def ossRHUsername = System.getenv("CONCLAVE_OSSRH_USERNAME")
+def ossRHPassword = System.getenv("CONCLAVE_OSSRH_PASSWORD")
+
 ext {
     conclave_revision = getConclaveRevision()
     project.logger.info("Conclave revision: $conclave_revision")
+
+    if (version.endsWith("-SNAPSHOT")) {
+        artifactType = ArtifactType.SNAPSHOT
+    } else if (version.matches(/.+\-RC[0-9]+$/)) {
+        artifactType = ArtifactType.RELEASE_CANDIDATE
+    } else if ((gitBranch == "release/$conclave_version") && ossRHUsername != null) {
```

Comment:
The version string should be enough to determine what the version type is, no need to double-check the branch.

Reply:
This is something I had issues with. I.e. when I was playing locally I had a current `release/1.3` branch (by mistake) and I had issues as the build was trying to run the sign task.
I also thought about the possibility where we branch out from the current `release/1.3` and push the changes. In that case, again, we would have issues as the build would try to sign and publish.
The checks above are narrowing the possibility of mistakes like mine

Reply:
We shouldn't be branching off release/1.3. Why would we do that? Patch releases (e.g. 1.3.1) will continue to be on release/1.3.

I'm not comfortable with this check. It changes the defintion of the version based on the branch and the local setup (i.e. ossRHUsername).